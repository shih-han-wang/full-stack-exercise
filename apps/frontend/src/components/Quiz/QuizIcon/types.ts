export enum IconVarient {
  Correct = 'correct',
  Incorrect = 'incorrect',
  Current = 'current',
  Default = 'default'
}
