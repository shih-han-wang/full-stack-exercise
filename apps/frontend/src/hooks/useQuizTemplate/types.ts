export enum Template {
  Number = 'number',
  RadioGrope = 'radio_group',
  Text = 'text'
}
