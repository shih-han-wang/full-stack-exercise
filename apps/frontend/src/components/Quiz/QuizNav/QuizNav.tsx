import { ChevronLeftIcon, ChevronRightIcon } from '@chakra-ui/icons'
import { Button, Flex } from '@chakra-ui/react'
import { useRouter } from 'next/router'
import { QuizQuestion } from '../../../types/query/quiz.types'
import QuizIcon from '../QuizIcon/QuizIcon'

type Props = {
  questions: QuizQuestion[]
  currentOrder: number
}

const QuizNav = ({ questions, currentOrder }: Props) => {
  const router = useRouter()
  const handleOnClick = (id: string) => id && router.push(`/question/${id}`)

  const nextQuestion = questions?.find(
    ({ order }) => order === currentOrder + 1
  )
  const currentQuestion = questions?.find(({ order }) => order === currentOrder)
  const previousQuestion = questions?.find(
    ({ order }) => order === currentOrder - 1
  )

  return (
    <Flex minWidth='100%' alignItems='center' justifyContent='center'>
      <Flex justifyContent='flex-start' flex='1'>
        {!!previousQuestion && (
          <Button
            size='md'
            colorScheme='gray'
            variant='outline'
            onClick={() => handleOnClick(previousQuestion?.id)}
          >
            <ChevronLeftIcon />
            Back
          </Button>
        )}
      </Flex>
      <Flex gap='2'>
        {questions.map((question) => (
          <QuizIcon
            key={question.id}
            question={question}
            currentOrder={currentOrder}
          />
        ))}
      </Flex>
      <Flex justifyContent='flex-end' flex='1'>
        {nextQuestion && (
          <Button
            size='md'
            colorScheme='gray'
            variant='outline'
            onClick={() => handleOnClick(nextQuestion?.id)}
            disabled={!currentQuestion?.response_correctness}
          >
            <ChevronRightIcon />
            Next
          </Button>
        )}
      </Flex>
    </Flex>
  )
}

export default QuizNav
