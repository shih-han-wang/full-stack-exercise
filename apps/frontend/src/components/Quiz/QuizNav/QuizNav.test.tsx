import { render } from '@testing-library/react'
import { Quizzes_Questions_Response_Correctness_Enum } from 'backend/sdk/generated'
import { QuizQuestion } from '../../../types/query/quiz.types'
import QuizNav from './QuizNav'

const mockQuestions: QuizQuestion[] = [
  {
    __typename: 'quizzes_questions',
    id: 'id-1',
    order: 1,
    response_correctness: Quizzes_Questions_Response_Correctness_Enum.Correct
  },
  {
    __typename: 'quizzes_questions',
    id: 'id-2',
    order: 2,
    response_correctness: Quizzes_Questions_Response_Correctness_Enum.Incorrect
  },
  {
    __typename: 'quizzes_questions',
    id: 'id-3',
    order: 3,
    response_correctness: null
  },
  {
    __typename: 'quizzes_questions',
    id: 'id-4',
    order: 4,
    response_correctness: null
  }
]

describe('Components > Quiz > QuizNav', () => {
  it('should render correctly', async () => {
    const view = render(<QuizNav questions={mockQuestions} currentOrder={3} />)

    expect(view.asFragment()).toMatchSnapshot()
  })
})
