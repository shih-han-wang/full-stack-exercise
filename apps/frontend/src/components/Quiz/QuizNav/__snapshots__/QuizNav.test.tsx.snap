// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Quiz > QuizNav should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <div
      class="css-0"
    >
      <button
        class="chakra-button css-0"
        type="button"
      >
        <svg
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            fill="currentColor"
          />
        </svg>
        Back
      </button>
    </div>
    <div
      class="css-0"
    >
      <div
        class="css-0"
      >
        <svg
          class="chakra-icon css-19pt3uk"
          focusable="false"
          viewBox="0 0 14 14"
        >
          <g
            fill="currentColor"
          >
            <polygon
              points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
            />
          </g>
        </svg>
      </div>
      <div
        class="css-0"
      >
        <svg
          class="chakra-icon css-19pt3uk"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="css-0"
      >
        <svg
          class="chakra-icon css-19pt3uk"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="css-0"
      >
        <svg
          class="chakra-icon css-19pt3uk"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-0"
    >
      <button
        class="chakra-button css-0"
        disabled=""
        type="button"
      >
        <svg
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            fill="currentColor"
          />
        </svg>
        Next
      </button>
    </div>
  </div>
</DocumentFragment>
`;
